specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2019-09-01/keys.json

```diff
@@ -0,0 +1,588 @@
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/CloudErrorBody"
+        }
+      },
+      "x-ms-external": true
+    },
+    "CloudErrorBody": {
+      "description": "An error response from Key Vault resource provider",
+      "properties": {
+        "code": {
+          "type": "string",
+          "description": "Error code. This is a mnemonic that can be consumed programmatically."
+        },
+        "message": {
+          "type": "string",
+          "description": "User friendly error message. The message is typically localized and may vary with service version."
+        }
+      },
+      "x-ms-external": true
+    },
+    "Attributes": {
+      "properties": {
+        "enabled": {
+          "type": "boolean",
+          "description": "Determines whether or not the object is enabled."
+        },
+        "nbf": {
+          "x-ms-client-name": "NotBefore",
+          "type": "integer",
+          "format": "int64",
+          "description": "Not before date in seconds since 1970-01-01T00:00:00Z."
+        },
+        "exp": {
+          "x-ms-client-name": "Expires",
+          "type": "integer",
+          "format": "int64",
+          "description": "Expiry date in seconds since 1970-01-01T00:00:00Z."
+        },
+        "created": {
+          "type": "integer",
+          "format": "int64",
+          "readOnly": true,
+          "description": "Creation time in seconds since 1970-01-01T00:00:00Z."
+        },
+        "updated": {
+          "type": "integer",
+          "format": "int64",
+          "readOnly": true,
+          "description": "Last updated time in seconds since 1970-01-01T00:00:00Z."
+        },
+        "recoveryLevel": {
+          "type": "string",
+          "description": "The deletion recovery level currently in effect for the object. If it contains 'Purgeable', then the object can be permanently deleted by a privileged user; otherwise, only the system can purge the object at the end of the retention interval.",
+          "enum": [
+            "Purgeable",
+            "Recoverable+Purgeable",
+            "Recoverable",
+            "Recoverable+ProtectedSubscription"
+          ],
+          "x-ms-enum": {
+            "name": "DeletionRecoveryLevel",
+            "modelAsString": true
+          },
+          "readOnly": true,
+          "x-nullable": false
+        }
+      },
+      "description": "The object attributes managed by the Azure Key Vault service."
+    },
+    "KeyProperties": {
+      "properties": {
+        "attributes": {
+          "$ref": "#/definitions/KeyAttributes",
+          "description": "The attributes of the key."
+        },
+        "kty": {
+          "type": "string",
+          "minLength": 1,
+          "description": "The type of the key. For valid values, see JsonWebKeyType.",
+          "enum": [
+            "EC",
+            "EC-HSM",
+            "RSA",
+            "RSA-HSM"
```

Comment:
Not "oct" for Managed HSM?

Reply:
Key Vault doesn't currently support "oct", but perhaps it's a good idea to include this in the enum anyways in case we eventually do. I'll add it.